### ISSUE TITLE
How would I ingest jsonl?

### ISSUE THREAD

**Issue by springcoil** [FreeDiscovery/FreeDiscovery#187]
I have a single file with json.
In pandas this looks like this 
- ```

content | id | media-type | published | source | title
-- | -- | -- | -- | -- | --
By Ryan Mayer, CBS Local Sports\n\nWeek 1 of t... | ad1dfe61-164b-4fd8-8ef3-38d676f41831 | Blog | 2015-09-17T17:28:13Z | CBS Baltimore | Thursday Night Football Q&A: Tiki Barber "Peyt...
© \r\n Android Pay\r \nGoogle has made its mo... | 383864ab-4880-46d0-8625-44ab43f489bc | News | 2015-09-11T00:15:15Z | Malaysia MSN | Google takes on Apple with Android Pay
For some time now, I've noticed that one of my... | 36c044d8-4522-4a28-a92c-a8e994742188 | Blog | 2015-09-14T13:29:49Z | Jarred at the movies | Movie of the Week: Black Mass
The article below represents a preview only an... | 374ff291-51c3-453e-ab06-c18dd2d880a0 | News | 2015-09-10T21:01:53Z | HT Syndication | C'garh decked for Modi's visit
Environment groups urge new Prime Minister Mal... | c463033c-21a8-4085-ba81-f894ab0ddf5d | News | 2015-09-20T06:03:08Z | Wauchope Gazette | Green groups urge Malcolm Turnbull to drop Ton...
```
How would I import that to free discovery?

**Comment by rth** [comment]
Hi @springcoil ,

it's a good question. In Freediscovery you would currently have to manually concatenate e.g; `content, media-type, source` and `title` fields, in pandas, and write those as one text file per row in some folder that can be ingested in freediscovery. That's not ideal and better support for other input formats is definitely on the roadmap https://github.com/FreeDiscovery/FreeDiscovery/issues/154

Though in this case, even assuming no implementation constraints, it's not fully clear what should be optimal solution. For instance, using [`ColumnTransformer`](http://scikit-learn.org/stable/modules/generated/sklearn.compose.ColumnTransformer.html) from scikit-learn 0.20 we could e.g. apply text vectorizers to `content` and `title` while doing one-hot-encoding of `media-type` and `source`, but it's unclear what would be a good feature weighting (e.g. TFIDF+L2 for text, L2 for the rest) before concatenation to preserve meaningful cosine distances between documents.

**Comment by springcoil** [comment]
That'd be quite cumbersome. I might have a go later

**Comment by rth** [comment]
Yes, a better solution might be to extract features from your jsonl manually with scikit-learn and `ColumnTranformer`, then provide the resulting sparse CSR matrix of features to FreeDiscovery and do the rest of the processing there.

If you can provide a link to the subset of your data, I can try to come up with an example...
